assets/check

```diff
@@ -37,7 +38,9 @@ configure_git_global "${git_config_payload}"
 destination=$TMPDIR/git-resource-repo-cache
 
 tagflag=""
-if [ -n "$tag_filter" ]; then
+if [ "$fetch_tags" == "false" ] ; then
+  tagflag="--no-tags"
+elif [ -n "$tag_filter" ] || [ "$fetch_tags" == "true" ] ; then
   tagflag="--tags"
```

Comment:
Hmm...if `source.fetch_tags` is intended to set the default behaviour, I'm realizing now that setting `source.fetch_tags` to `false` and defining `source.tag_filter` may not be such an unusual case (whenever you clone the repo in a `get` step, you don't care about the tags, but you want to trigger jobs for only commits that match some `tag_filter`).

With the logic the way it is, however, this won't actually work - since `$fetch_tags` is `"false"`, we always clone the repo with `--no-tags`. When we later try to run `git tag --list "$tag_filter"`, we'll always end up with an empty list, meaning the resource won't produce any versions.

I wonder if we even need to consider `source.fetch_tags` in the `check` script...we only ever need to fetch tags when `tag_filter` is set. Perhaps this could become the following:

```suggestion
if [ -n "$tag_filter" ]; then
  tagflag="--tags"
else
  tagflag="--no-tags"
```

Note that this would only be for `check` (i.e. not `in`).

Does that make sense? Let me know if what I'm saying won't work.

Reply:
I had a look on `check` and I did some test with `--no-tag`. Tags are needed only when using `tag_filter`.

As you said, we should be able to remove `fetch_tags` from `check` and set `--no-tags` as default behavior.

I will do more tests next week before pushing the changes.

Reply:
Awesome, and thanks for being so responsive!

Reply:
I did more tests on the `check` no issue discovered. I think we are fine ;)